http/src/request/attachment.rs

```diff
@@ -1,30 +1,183 @@
+use crate::request::Form;
+use serde::{Deserialize, Serialize};
+use twilight_model::id::{marker::AttachmentMarker, Id};
+
+pub struct AttachmentManager<'a> {
+    files: Vec<&'a Attachment>,
+    ids: Vec<Id<AttachmentMarker>>,
+}
+
+impl<'a> AttachmentManager<'a> {
+    pub const fn new() -> Self {
+        Self {
+            files: Vec::new(),
+            ids: Vec::new(),
+        }
+    }
+
+    pub fn build_form(&self, fields: &'a [u8]) -> Form {
+        let mut form = Form::new().json_part(b"payload_json", fields);
+
+        for (index, file) in self.files.iter().enumerate() {
+            let mut name = Vec::with_capacity(7 + num_digits(index));
+            name.extend(b"files[");
+            push_digits(index as u64, &mut name);
+            name.extend(b"]");
+
+            form = form.file_part(name.as_ref(), file.filename.as_bytes(), file.file.as_ref())
+        }
+
+        form
+    }
+
+    pub fn get_partial_attachments(&self) -> Vec<PartialAttachment<'a>> {
+        self.files
+            .iter()
+            .enumerate()
+            .map(|(index, attachment)| PartialAttachment {
+                description: attachment.description.as_deref(),
+                filename: Some(attachment.filename.as_ref()),
+                id: index as u64,
+            })
+            .chain(self.ids.iter().map(|id| PartialAttachment {
+                description: None,
+                filename: None,
+                id: id.get(),
+            }))
+            .collect()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.files.is_empty() && self.ids.is_empty()
+    }
+
+    pub fn set_files(mut self, files: Vec<&'a Attachment>) -> Self {
+        self.files = files;
+
+        self
+    }
+
+    pub fn set_ids(mut self, ids: Vec<Id<AttachmentMarker>>) -> Self {
+        self.ids = ids;
+
+        self
+    }
+}
+
 /// Attachment for when creating and updating messages.
 #[derive(Clone, Debug)]
-pub struct AttachmentFile<'a> {
-    pub(crate) filename: &'a str,
-    pub(crate) description: Option<&'a str>,
-    pub(crate) file: &'a [u8],
+pub struct Attachment {
+    pub description: Option<String>,
+    pub file: Vec<u8>,
+    pub filename: String,
```

Comment:
Is there a reason this is now a owned struct, I could see it lead to less optimal memory use if you already have what you are sending in memory or as a static in the file

Reply:
It's owned to avoid putting a lifetime on every struct that uses it. I think since this probably doesn't make it in to the next major release we can spend some time rectifying that in the future

Reply:
If you don't have any comments on this, I'll resolve and request another review